datafusion/proto/tests/cases/roundtrip_logical_plan.rs

```diff
@@ -622,8 +622,8 @@ async fn roundtrip_expr_api() -> Result<()> {
         ),
         array_replace_all(make_array(vec![lit(1), lit(2), lit(3)]), lit(2), lit(4)),
         first_value(vec![lit(1)], false, None, None, None),
-        covar_samp(lit(1.5), lit(2.2), false, None, None, None),
```

Comment:
This change now means the expr_fn can't be used with ordering / null_treatment right? I think that is probably ok

Perhaps we can consider some better API for creating aggregate functions with `ORDER BY`, perhaps inspired by https://github.com/apache/datafusion/issues/6747

Perhaps something like 
```rust
// form `FIRST_VALUE(a ORDER BY b)`
let agg_expr = AggregateUDF::call(first_value())
  .args(col("a"))
  .order_by(vec![col("b"))
  .build()?;
```

Reply:
> This change now means the expr_fn can't be used with ordering / null_treatment right? I think that is probably ok
> 
> Perhaps we can consider some better API for creating aggregate functions with `ORDER BY`, perhaps inspired by #6747
> 
> Perhaps something like
> 
> ```rust
> // form `FIRST_VALUE(a ORDER BY b)`
> let agg_expr = AggregateUDF::call(first_value())
>   .args(col("a"))
>   .order_by(vec![col("b"))
>   .build()?;
> ```

This looks nice